### PR TITLE
ENT-2790: Port CrlResource to spec-first

### PR DESCRIPTION
**Pull request by Januson**

### CODE REVIEW

File: api/candlepin-api-spec.yaml [hunk 1]
```diff
@@ -1215,6 +1215,64 @@ paths:
...
+              example:
+                displayMessage: CRL file path not defined in config file
+                requestUuid: c4347004-8792-41fe-a4d8-fccaa0d3898a
+        default:
+          $ref: '#/components/responses/default'
+
+    delete:
+      tags:
+        - Certificate Revocation List
+      description: Deletes a Certificate from the Revocation List
```
Comment (sonalidhome) [inline api/candlepin-api-spec.yaml:1250]: The same applies here. Let's avoid capitalization of the display message.

File: api/candlepin-api-spec.yaml [hunk 2]
```diff
@@ -1215,6 +1215,64 @@ paths:
...
+        - Certificate Revocation List
+      description: Retrieves the Certificate Revocation List
+      operationId: getCurrentCrl
+      security: []
+      x-java-response:
+        type: javax.ws.rs.core.Response
+        isContainer: false
+      responses:
+        200:
+          description: Certificate Revocation List successfully retrieved
```
Comment (sonalidhome) [inline api/candlepin-api-spec.yaml:1230]: Let's avoid capitalization of the display message.

File: server/src/main/java/org/candlepin/resource/CrlResource.java [hunk 3]
```diff
@@ -26,55 +25,44 @@
...
-import javax.ws.rs.QueryParam;
-import javax.ws.rs.core.Context;
-import javax.ws.rs.core.MediaType;
 import javax.ws.rs.core.Response;
 
 /**
  * CrlResource
  */
 @Path("/crl")
 @Api(value = "crl", authorizations = { @Authorization("basic") })
```
Comment (sonalidhome) [inline server/src/main/java/org/candlepin/resource/CrlResource.java:46]: `@Path` and `@Api` annotations can be removed.

**Review by sonalidhome — CHANGES_REQUESTED** [review]

File: server/src/main/java/org/candlepin/resource/CrlResource.java (hunk 3 above)
Comment (Januson) [reply server/src/main/java/org/candlepin/resource/CrlResource.java]: Done.

File: api/candlepin-api-spec.yaml (hunk 2 above)
Comment (Januson) [reply api/candlepin-api-spec.yaml]: Done.

File: api/candlepin-api-spec.yaml (hunk 1 above)
Comment (Januson) [reply api/candlepin-api-spec.yaml]: Done.

**Review by sonalidhome — APPROVED** [review]

**Pull request merged**